### PR TITLE
[Unity] Legalization for LayoutTransform

### PR DESCRIPTION
**Pull request by LeshengJin**
This pr implements layout_transform legalization mentioned in https://discuss.tvm.apache.org/t/codegen-support-for-relax-layout-transform/15187.

### CODE REVIEW

File: tests/python/relax/test_transform_legalize_ops_manipulate.py [hunk 1]
```diff
@@ -1522,5 +1522,45 @@ def main(
...
 
 
+def test_layout_transform():
+    transformation = lambda a, b, c: (a, c, b // 3, b % 3)
+    pad_value = 2
+    # fmt: off
+    @I.ir_module
+    class LayoutTransform:
+        @R.function
+        def main(x: R.Tensor((10, 20, 30), "float32")):
```
Comment (psrivas2) [inline tests/python/relax/test_transform_legalize_ops_manipulate.py:1532]: should we also add a test case with symbolic dimensions?

**Review by psrivas2 — APPROVED** [review]
Looks good! Thanks @LeshengJin !

File: tests/python/relax/test_transform_legalize_ops_manipulate.py (hunk 1 above)
Comment (LeshengJin) [reply tests/python/relax/test_transform_legalize_ops_manipulate.py:1532]: Good idea! Have added a symbolic shape test case.

File: python/tvm/relax/transform/legalize_ops/manipulate.py [hunk 2]
```diff
@@ -163,3 +163,30 @@ def _scatter_elements(bb: BlockBuilder, call: Call) -> Expr:
...
+def _layout_transform(bb: BlockBuilder, call: Call) -> Expr:
+    index_map: tvm.tir.IndexMap = call.attrs.index_map
+    pad_value = call.attrs.pad_value.value
+
+    def te_layout_transform(data):
+        inverse, padding_predicate = index_map.non_surjective_inverse(data.shape)
+        output_shape = index_map.map_shape(data.shape)
+        return te.compute(
+            output_shape,
+            lambda *idx: tvm.te.if_then_else(
```
Comment (vinx13) [inline python/tvm/relax/transform/legalize_ops/manipulate.py:178]: shall we simplified if_then_else when the predicate is not needed?

Comment (LeshengJin) [reply python/tvm/relax/transform/legalize_ops/manipulate.py]: Yes, that's better. Added an if-else statement to check if the predicated is needed.

**Review by Hzfengsy — APPROVED** [review]

**Pull request merged**